walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,125 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // Save all attendees that should attend and the duration of the meeting.
+    Collection<String> attendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    long duration = request.getDuration();
+
+    // Add all time ranges that someone who is in the meeting will be in to the correct
+    // mandatory, and optional and mandatory time range arrayLists.
+    ArrayList<TimeRange> mandatoryTimeRanges = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> optionalMandatoryTimeRanges = new ArrayList<TimeRange>();
+    Iterator<Event> eventsIterator = events.iterator();
+    while (eventsIterator.hasNext()) {
+      Event event = eventsIterator.next();
```

Comment:
just want to double-check, this logic won't skip the very first item in events, right?

Reply:
I was confused by the way this works at first too. It won't skip the first element. Here is an [example](https://www.tutorialspoint.com/iterate-through-arraylist-in-java#:~:text=Some%20of%20the%20important%20methods,next%20element%20in%20the%20ArrayList.).